app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -836,15 +824,8 @@ watchEffect(() => {
       : dataHeader
 
     if (!data_.is_using_server_sort_and_filter) {
-      const hasIndexRow =
-        config.nodeType === TABLE_NODE_TYPE ||
-        config.nodeType === COLUMN_NODE_TYPE ||
-        config.nodeType === DB_TABLE_NODE_TYPE
-      const shift = hasIndexRow ? 1 : 0
-      rowData.value =
-        data_.data ?
-          createRowsForTable(data_.data, shift, data_.is_using_server_sort_and_filter)
-        : []
+      const shift = data_.type === 'EnsoTableOrColumn' ? 1 : 0
```

Comment:
Defining a type predicate for `EnsoTableOrColumn` would reduce the risk that this check becomes inconsistent with the types in the future